spec/features/admin/merchants/index_spec.rb

```diff
@@ -85,5 +83,218 @@
+          5.times do |i|
+            expect(actual_ordered_text[i]).to include(expected_ordered_names[i])
+          end
+        end
+      end
+
+      it 'merchant names are links to their show pages' do
+        setup_top_five_revenue_merchants
+        visit admin_merchants_path
+
+        within('#top-merchants') do
+          actual_ordered_links = page.all('li a').map { |a| a['href'] }
+          expected_ordered_links = [admin_merchant_path(@merchant_5),
+                                    admin_merchant_path(@merchant_4),
+                                    admin_merchant_path(@merchant_3),
+                                    admin_merchant_path(@merchant_2),
+                                    admin_merchant_path(@merchant_1)]
+
+          expect(actual_ordered_links).to eq(expected_ordered_links)
+        end
+      end
+
+      it 'shows total revenue next to merchant name' do
+        setup_top_five_revenue_merchants
+        visit admin_merchants_path
+
+        within('#top-merchants') do
+          actual_ordered_text = page.all('.top-merchant').map(&:text)
+          expected_currency_strings = ["$#{@merchant_5.total_revenue.truncate}",
+                                            "$#{@merchant_4.total_revenue.truncate}",
+                                            "$#{@merchant_3.total_revenue.truncate}",
+                                            "$#{@merchant_2.total_revenue.truncate}",
+                                            "$#{@merchant_1.total_revenue.truncate}"]
+
+          expect(actual_ordered_text.size).to eq(5)
+          5.times do |i|
+            expect(actual_ordered_text[i]).to include(expected_currency_strings[i])
+          end
+        end
+      end
+
+      it 'shows date with most revenue for each merchant' do
+        merchant_8_with_history
+        merchant_9_with_history
+        merchant_10_with_history
+        visit admin_merchants_path
+
+        within('#top-merchants') do
+          actual_best_day_text = page.all('.top-merchant-best-day').map(&:text)
+          expected_best_day_strings = [@merchant_8.best_day.strftime('%A, %B %d, %Y'),
+                                        @merchant_9.best_day.strftime('%A, %B %d, %Y'),
+                                        @merchant_10.best_day.strftime('%A, %B %d, %Y')]
+
+          expect(actual_best_day_text.size).to eq(3)
+          3.times do |i|
+            expect(actual_best_day_text[i]).to include(expected_best_day_strings[i])
+          end
+        end
+      end
+    end
+  end
+
+  def sample_date
+    DateTime.new(2021, 01, 01)
+  end
+
+  def setup_basic_merchants
+    @merchant_11 = create(:merchant, status: :enabled)
+    @merchant_12= create(:merchant, status: :enabled)
+    @merchant_13 = create(:merchant, status: :enabled)
+    @merchant_14, @merchant_15, @merchant_16 = create_list(:merchant, 3)
+  end
+
+  def setup_top_five_revenue_merchants
+    merchant_1_with_history
```

Comment:
This test setup is so organized, great idea to make this lengthy file work so well.